### PR TITLE
Feature: Delete button and summary section

### PR DESCRIPTION
**Pull request by Long-HaHoang**
- Creating a button in the shopping cart item card
- Filter (Delete) items from the shopping cart
- State management
- Summary section for total price, item amount and shipping cost

### CODE REVIEW

File: components/Gallery/index.js [hunk 1]
```diff
@@ -19,7 +19,11 @@ export default function Gallery({ onCartItem }) {
     <Styled.GallerySection>
       <Styled.GalleryList>
         {isLoading ? (
-          <p>Is Loading...</p>
+          <>
+            <br />
+            <Styled.LoadingCube />
+            <br />
+          </>
```
Comment (doemser) [inline components/Gallery/index.js:26]:
```suggestion

            <Styled.LoadingCube />

```

File: components/Icons/SmallArrowLeft.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+      strokeWidth={1.5}
+      stroke="currentColor"
+      className="w-6 h-6"
+    >
+      <path
+        strokeLinecap="round"
+        strokeLinejoin="round"
+        d="M10.5 19.5L3 12m0 0l7.5-7.5M3 12h18"
+      />
+    </StyledButtonSVG>
```
Comment (doemser) [inline components/Icons/SmallArrowLeft.js:19]: You are handling SVG icons kind of unfavorable, see if this approach can help you to more order:

https://codesandbox.io/s/dreamy-joana-13ij6o?file=/src/App.js

File: components/Layout/Layout.styled.js [hunk 3]
```diff
@@ -1,14 +1,24 @@
...
   justify-content: center;
   align-items: center;
   border: solid green;
-  padding: 0 10px;
-  gap: 10;
+  padding: 20px 0;
 `;
 
-export { AppContainer };
+const Footer = styled.div`
```
Comment (doemser) [inline components/Layout/Layout.styled.js:14]:
```suggestion
const Footer = styled.footer`
```

File: components/ShopHeader/index.js [hunk 4]
```diff
@@ -14,9 +14,14 @@ export default function ShopHeader({ cartTotal }) {
...
     <Styled.ShopHeader>
       <Styled.ShopH1>Shop Name</Styled.ShopH1>
-      <Styled.ShoppingCartLink href={"/ShoppingCart"}>
+      <Styled.ShoppingCartLink
+        aria-label="Checkout link"
+        href={"/ShoppingCart"}
+      >
         <SmallCart />
-        <span>{cartTotal}</span>
+        {cartTotal !== 0 ? (
```
Comment (doemser) [inline components/ShopHeader/index.js:22]:
```suggestion
        {cartTotal ? (
```

?

File: styles.js [hunk 5]
```diff
@@ -7,12 +7,20 @@ export default createGlobalStyle`
...
+    --secondary-color: #0c0c0c;
+    
     margin: 0;
-    font-family: system-ui;
+    font-family: Roboto, sans-serif;
     display: flex;
     justify-content: center;
-    border: solid red;
+
+    /* border: solid red; */
```
Comment (doemser) [inline styles.js:21]: no outcommented code

**Review by doemser — CHANGES_REQUESTED** [review]
Just a few comments. You could well switch to a global state solution like jotai or zutsand.js to stop drilling props from _app throughout your app.

I haven't looked at your stylings in detail now, QA can do that.

File: components/Gallery/index.js (hunk 1 above)
Comment (Long-HaHoang) [reply components/Gallery/index.js:26]: This one is on my to-do will refactor the LoadingCube in the next story.
There are some edge case visual bugs I had to fix.
I already got a solution for it.

File: components/Icons/SmallArrowLeft.js (hunk 2 above)
Comment (Long-HaHoang) [reply components/Icons/SmallArrowLeft.js:19]: I will look into it next when I do a styling overhaul.
I know about variants but I don't want to concentrate my time on icons, yet.

File: components/Layout/Layout.styled.js (hunk 3 above)
Comment (Long-HaHoang) [reply components/Layout/Layout.styled.js:14]: - [x] Already fixed forgot to push it. Thanks for the reminder. 👍

File: styles.js (hunk 5 above)
Comment (Long-HaHoang) [reply styles.js:21]: - [x] Fixed

File: components/ShopHeader/index.js (hunk 4 above)
Comment (Long-HaHoang) [reply components/ShopHeader/index.js:22]: - [x] Kind of makes more sense, thanks. Fixed.

File: components/Gallery/index.js (hunk 1 above)
Comment (Long-HaHoang) [reply components/Gallery/index.js:26]: - [x] Fixed with a margin for now.

File: components/ShopHeader/index.js [hunk 6]
```diff
@@ -14,9 +14,14 @@ export default function ShopHeader({ cartTotal }) {
...
-      <Styled.ShoppingCartLink href={"/ShoppingCart"}>
+      <Styled.ShoppingCartLink
+        aria-label="Checkout link"
+        href={"/ShoppingCart"}
+      >
         <SmallCart />
-        <span>{cartTotal}</span>
+        {cartTotal ? (
+          <Styled.ShoppingCartCounter>{cartTotal}</Styled.ShoppingCartCounter>
+        ) : null}
```
Comment (NikRoe) [inline components/ShopHeader/index.js:24]: even shorter and easier to read:

```suggestion
        {cartTotal && <Styled.ShoppingCartCounter>{cartTotal}</Styled.ShoppingCartCounter>}
```

File: components/ShoppingCart.styled.js [hunk 7]
```diff
@@ -21,7 +22,98 @@ const CartItemList = styled.ul`
 `;
 
 const TopInfoSection = styled.section`
-  border: solid black;
+  position: relative;
+  border-bottom: solid lightgray;
+  width: 100%;
+  padding: 0px 10px;
```
Comment (NikRoe) [inline components/ShoppingCart.styled.js:28]: you don't need a unit here

```suggestion
  padding: 0 10px;
```

File: components/ShoppingCart.styled.js [hunk 8]
```diff
@@ -21,7 +22,98 @@ const CartItemList = styled.ul`
...
+  width: 100%;
+  display: flex;
+  flex-direction: row;
+  justify-content: space-evenly;
+`;
+
+const SummarySection = styled.div`
+  position: relative;
+  width: 100%;
+  padding: 0px 10px;
```
Comment (NikRoe) [inline components/ShoppingCart.styled.js:64]:
```suggestion
  padding: 0 10px;
```

**Review by NikRoe — CHANGES_REQUESTED** [review]
Hi there,

looks very good and I left only some minor remarks.

File: components/ShopHeader/index.js (hunk 6 above)
Comment (Long-HaHoang) [reply components/ShopHeader/index.js:24]: -[x] Good suggestion.

File: components/ShoppingCart.styled.js (hunk 7 above)
Comment (Long-HaHoang) [reply components/ShoppingCart.styled.js:28]: - [x] Fixed

File: pages/ShoppingCart/index.js [hunk 9]
```diff
@@ -22,28 +22,60 @@ export default function ShoppingCartPage({ cartItems }) {
...
+            <hr />
+            <Styled.SummaryParagraph>
+              <span>Total price: </span>
+              <span>
+                {formatNumberToDeCurrency(sumOfAllItemPrices + shippingCost)}
+              </span>
+            </Styled.SummaryParagraph>
+          </Styled.SummaryCard>
+        </Styled.SummarySection>
+      ) : null}
```
[the remark this answers is not in the document]
Comment (Long-HaHoang) [reply pages/ShoppingCart/index.js:78]: - [x] Good one, thanks 👍

**Pull request merged**